library/collect_container_configs

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python2.7
+
+import json
+import os
+import os.path
+
+DOCUMENTATION = '''
+---
+module: collect_container_configs
+short_description: collects container configuration data
+description:
+  - when run, this module collects the contents of container configuration files for
+    all current containers
+  - this can be run before an upgrade to store information that would be used after
+    an upgrade
+'''
+
+
+CONTAINER_ROOT_DIR = '/var/lib/docker/containers'
+CONFIG_V1_FILENAME = 'config.json'
+CONFIG_V2_FILENAME = 'config.v2.json'
+CONFIG_V3_FILENAME = 'config.v3.json'
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(),
+        supports_check_mode=True)
+
+    configs = {}
+    config_versions = {}
+
+    for container in os.listdir(CONTAINER_ROOT_DIR):
+        container_path = os.path.join(CONTAINER_ROOT_DIR, container)
+        if not os.path.isdir(container_path):
+            # we only expect directories here. Ignore the unexpected
+            continue
+
+        config_path = os.path.join(container_path, CONFIG_V3_FILENAME)
+        if os.path.isfile(config_path):
+            version = 3
+        else:
+            config_path = os.path.join(container_path, CONFIG_V2_FILENAME)
+            if os.path.isfile(config_path):
+                version = 2
+            else:
+                config_path = os.path.join(container_path, CONFIG_V1_FILENAME)
+                if not os.path.isfile(config_path):
+                    continue
+                version = 1
```

Comment:
To support use of the filenames by version, below, one could reverse the keys and values (and their names in the _for_ loop) in my preceding suggestion, then refer to them like `CONFIG_FILENAME[2]` as needed.